nixos/modules/services/web-apps/pixelfed.nix

```diff
@@ -0,0 +1,366 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.pixelfed;
+  opt = options.services.pixelfed;
+
+  user = "pixelfed";
+  group = "nginx";
+
+  pixelfed = pkgs.pixelfed.override {
+    dataDir = cfg.dataDir;
+  };
+
+  configFile = pkgs.writeTextFile {
+    name = "env";
+    text = cfg.envFile + ''
+    APP_KEY = ${cfg.appKey}
+    DB_CONNECTION=${cfg.database.type}
+    DB_HOST=${cfg.database.host}
+    DB_PORT= ${toString cfg.database.port}
+    DB_DATABASE=${cfg.database.name}
+    DB_USERNAME=${cfg.database.user}
+    DB_PASSWORD=${cfg.database.password}
+    '';
+  };
+
+  # shell script for local administration
+  artisan = pkgs.writeScriptBin "pixelfed" ''
+    #! ${pkgs.runtimeShell}
+    cd ${pixelfed}
+    sudo=exec
+    if [[ "$USER" != ${user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${user}'
+    fi
+    $sudo ${pkgs.php}/bin/php artisan $*
+  '';
+
+
+in {
+  options.services = {
+    pixelfed = {
+      enable = mkEnableOption (lib.mdDoc "the pixelfed service");
+
+      envFile = mkOption {
+        type = types.str;
```

Comment:
It would be great to have some RFC42-style parameters here using a env generator